components/resources/CommentSection.jsx

```diff
@@ -1,33 +1,30 @@
-import Container from "@mui/material/Container";
-import { useOperation } from "react-openapi-client";
+import { Alert, Container } from "@mui/material";
 import Loader from "../common/Loader";
 import Comment from "./Comment";
+import NewResourceComment from "./NewResourceComment";
+import useFetch from "../../hooks/use-fetch";
 
 function CommentSection({ resourceId }) {
-  const { loading, data, error } = useOperation(
-    "getResourceComments",
-    resourceId
+  const { data, error, mutate } = useFetch(
+    `/api/resources/${resourceId}/resource_comments`
   );
 
-  if (loading) {
-    return <Loader />;
-  }
+  if (error) return <Alert severity="error">Error</Alert>;
+  if (!data) return <Loader />;
 
-  if (error) {
-    return <div>Error</div>;
-  }
+  const newCommentCreation = () => {
+    mutate();
+  };
 
   return (
-    <Container maxWidth="sm">
+    <Container maxWidth="xl">
       {data.map((comment) => (
-        <Comment
-          key={comment.id}
-          userName={comment.user.name}
-          postTime={comment.created_at}
-          content={comment.content}
-          userId={comment.user.id}
-        />
+        <Comment key={comment.id} comment={comment} />
       ))}
+      <NewResourceComment
```

Comment:
I find it more user-friendly if the `new comment` button is above of the existing comments.